mautic/mautic
# New salesforce contacts DNC sync

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | Y
| Automated tests included? | N
| Issues addressed (#s or URLs) | 
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
We need to synchronize DNC record for new Lead/Contacts downloaded from SalesForce. Currently the plugin relied solely on DNC history, which is not synchronized unless we push something and nor it is synchronized if contact has been created *opted-out*.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:

1. Setting up SF plugin
2. Selecting “Use latest updated Do Not Contact record” under “Features” in Salesforce plugin
3. Pulling contacts from SF who are labeled as DNC out of the box into Mautic.
4. Contacts who were not already known contacts in Mautic and are labeled as DNC out of the box in SF will not show up a DNC in Mautic.


#### Steps to test this PR:
1. Load up this PR
2. Create SF contact opted-out
3. Sync
4. Verify downloaded is DNCed
